src/query/v2/plan/operator.cpp

```diff
@@ -247,6 +258,65 @@ class DistributedCreateNodeCursor : public Cursor {
     return requests;
   }
 
+  void PlaceNodesOnTheMultiFrame(MultiFrame &multi_frame, ExecutionContext &context) {
+    auto multi_frame_reader = multi_frame.GetValidFramesConsumer();
+    size_t i = 0;
+    MG_ASSERT(std::distance(multi_frame_reader.begin(), multi_frame_reader.end()));
+    for (auto &frame : multi_frame_reader) {
+      const auto primary_label = msgs::Label{.id = nodes_info_[0]->labels[0]};
```

Comment:
The first one is solved by reducing the possibility of providing multiple node creation infos. The reason is it is not clear how it should work. e.g.: the pulled multiframe has `n` valid frame, the cursor has `m` node infos, then what should happen? By reducing `m` to exactly one, it is clear: the same node creation info will be used on all valid frames in the multiframe. This doesn't mean it will create the same vertices, e.g.: `UNWIND [1,2] AS prop CREATE (n:label {property:prop})` will create nodes `(:label{property:1})` and `(:label {property:2})`.

For the issue about primary label I added an assertion.